chrome/content/zotero/xpcom/connector/server_connector.js

```diff
@@ -1143,7 +1143,11 @@ Zotero.Server.Connector.SaveSnapshot.prototype = {
 			return 500;
 		}
 		
-		return [201, "application/json", JSON.stringify({ saveSingleFile: !data.skipSnapshot })];
+		return [201,
+			"application/json",
+			JSON.stringify({
+				saveSingleFile: !data.skipSnapshot && !(data.pdf && library.filesEditable)
```

Comment:
Won't this be true if it's a PDF but the library doesn't have file editing enabled? `!(true && false) == true`

(It looks like in that case we now create an empty webpage item with the PDF URL, which is not ideal, but I'm not sure there's a better option. But we still don't want /saveSingleFile to be called.)